pkg/sloop/common/partition_utilities.go

```diff
@@ -0,0 +1,87 @@
+package common
+
+import (
+	"github.com/dgraph-io/badger/v2"
+	"github.com/golang/glog"
+	"github.com/salesforce/sloop/pkg/sloop/store/untyped/badgerwrap"
+	"sort"
+)
+
+type SloopKey struct {
+	TableName   string
+	PartitionID string
+}
+
+// returns TableName, PartitionId, error.
+func GetSloopKey(item badgerwrap.Item) (SloopKey, error) {
+	key := item.Key()
+	err, parts := ParseKey(string(key))
+	if err != nil {
+		return SloopKey{}, err
+	}
+
+	var tableName = parts[1]
+	var partitionId = parts[2]
+	return SloopKey{tableName, partitionId}, nil
+}
+
+type PartitionInfo struct {
+	TotalKeyCount          uint64
+	TableNameToKeyCountMap map[string]uint64
+}
+
+// prints all the keys histogram. It can help debugging when needed.
+func PrintKeyHistogram(db badgerwrap.DB) {
+	partitionTableNameToKeyCountMap, totalKeyCount := GetPartitionsInfo(db)
+	glog.Infof("TotalkeyCount: %v", totalKeyCount)
+
+	for partitionID, partitionInfo := range partitionTableNameToKeyCountMap {
+		for tableName, keyCount := range partitionInfo.TableNameToKeyCountMap {
+			glog.Infof("TableName: %v, PartitionId: %v, keyCount: %v", tableName, partitionID, keyCount)
+		}
+	}
+}
+
+// Returns the sorted list of partitionIDs from the given partitions Info map
+func GetSortedPartitionIDs(partitionsInfoMap map[string]*PartitionInfo) []string {
+	var sortedListOfPartitionIds []string
+
+	for partitionID, _ := range partitionsInfoMap {
+		sortedListOfPartitionIds = append(sortedListOfPartitionIds, partitionID)
+	}
+
+	// Sorted numbered strings here is ok since they are all of the same length
+	sort.Strings(sortedListOfPartitionIds)
+	return sortedListOfPartitionIds
+}
+
+// Gets the Information for partitions to key Count Map
+// Returns Partitions to KeyCount Map, Partitions TableName to Key Count and total key count
+func GetPartitionsInfo(db badgerwrap.DB) (map[string]*PartitionInfo, uint64) {
+	var totalKeyCount uint64 = 0
+	partitionIDToPartitionInfoMap := make(map[string]*PartitionInfo)
+
+	_ = db.View(func(txn badgerwrap.Txn) error {
+		iterOpt := badger.DefaultIteratorOptions
+		iterOpt.PrefetchValues = false
+		it := txn.NewIterator(iterOpt)
+		defer it.Close()
+		for it.Rewind(); it.Valid(); it.Next() {
+			item := it.Item()
+			sloopKey, err := GetSloopKey(item)
+			if err != nil {
+				glog.Infof("failed to parse information about key: %x", item.Key())
```

Comment:
when err != nil, should we continue to the next item?

Reply:
yes in case there is just one bad key which got added somehow. I would continue with the execution of main logic.

Reply:
if the only key is a bad key, the following logic should not work, and if you have 10000 keys, and 90% are bad, by adding `continue` you dont need to do the following calculation. Also your totalKeyCount should be valid key count. partitionIDToPartitionInfoMap should only contains valid key paritionID.

Reply:
added the continue